<?xml version="1.0" encoding="utf-8"?>
<launch>
	<arg name="veh" default="$(env VEHICLE_NAME)"/>
    <arg name="demo_name" value="lane_following"/>
	<arg name="output" value="log"/>
	<!-- start basic args -->
	<include file="$(find duckietown_demos)/launch/master.launch">
	    <!-- Basic arguments -->
		<arg name="veh" value="$(arg veh)"/>
        <arg name="demo_name" value="$(arg demo_name)"/>
		<arg name="param_file_name" value="default" />
		<arg name="visualization" value="true" />

	    <!-- Finite state machine >
	    <arg name="fsm" value="true"/>
	    <arg name="/fsm/logic_gate" value="false"/-->

		<!-- Camera and anti intagram -->
		<arg name="anti_instagram" value="true" />

		<!-- Deadreckoning (aka. Odometry)	-->
<!--		<arg name="/localization" value="true"/>
		<arg name="/localization/deadreckoning" value="true"/>
LP removing for now
-->
	    <!-- Lane Following stack -->
	    <arg name="lane_following" value="true"/>
	    <arg name="/lane_following/line_detection" value="true"/>
	    <arg name="line_detector_param_file_name" value="default" />
	    <arg name="/lane_following/ground_projection" value="true"/>
	    <arg name="/lane_following/lane_filter" value="true"/>
	    
  	</include>
	<!-- launch led emitter node with launch file -->
	<include file="$(find led_emitter)/launch/led_emitter_node.launch">
		<arg name="veh" value="$(arg veh)"/>

	</include>
	
	<!-- launching the rl_agent -->
	<group ns="$(arg veh)">
		<node pkg="rl_agent" type="deep_rl_node.py" name="rl_agent_node" output="screen">
			<param name="veh" value="$(arg veh)"/>
		</node>
	</group>
	
	
</launch>